### PR TITLE
[TA] Expose Length property again as now we have StringIndexType

### LINKED ISSUE #18018: [FEATURE REQ] Surface the property Length in textanalytics.
**Library or service name.**
What library or service is this request related to? 

TextAnalytics
https://github.com/Azure/azure-sdk-for-net/pull/17968#pullrequestreview-569576227 

**Is your feature request related to a problem? Please describe.**
What feature would you like to get added? What problem is it solving?

@maririos

### PR DESCRIPTION
**Pull request by maririos**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/18018

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Release History
 
 ## 5.1.0-beta.4 (Unreleased)
+### New features
+- Added property `Length` to `CategorizedEntity`, `SentenceSentiment`, `LinkedEntityMatch`, `AspectSentiment`, `OpinionSentiment`, and `PiiEntity`.
+- `StringIndexType` has been added to all endpoints as a new option to control how the `Offset` and `Length` are calculated by the service.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:6]: Missing from PR https://github.com/Azure/azure-sdk-for-net/pull/17968

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs [hunk 2]
```diff
@@ -498,10 +501,11 @@ public static LinkedEntityMatch LinkedEntityMatch(string text, double score)
         /// <param name="text">Sets the <see cref="LinkedEntityMatch.Text"/> property.</param>
         /// <param name="score">Sets the <see cref="LinkedEntityMatch.ConfidenceScore"/> property.</param>
         /// <param name="offset">Sets the <see cref="LinkedEntityMatch.Offset"/> property.</param>
+        /// <param name="length">Sets the <see cref="LinkedEntityMatch.Length"/> property.</param>
         /// <returns>A new instance of <see cref="TextAnalytics.LinkedEntityMatch"/> for mocking purposes.</returns>
-        public static LinkedEntityMatch LinkedEntityMatch(string text, double score, int offset)
+        public static LinkedEntityMatch LinkedEntityMatch(string text, double score, int offset, int length)
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs:506]: This method got introduced after GA so it is not a breaking change

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeSentimentTests.cs [hunk 3]
```diff
@@ -455,6 +455,7 @@ private void CheckAnalyzeSentimentProperties(DocumentSentiment doc, bool opinion
                         Assert.AreEqual(0, minedOpinions.Aspect.ConfidenceScores.Neutral);
                         Assert.IsTrue(CheckTotalConfidenceScoreValue(minedOpinions.Aspect.ConfidenceScores));
                         Assert.IsNotNull(minedOpinions.Aspect.Offset);
+                        Assert.IsNotNull(minedOpinions.Aspect.Length);
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeSentimentTests.cs:458]: Analyze sentiment tests need some love. Issue for that is here => https://github.com/Azure/azure-sdk-for-net/issues/14833

**Review by mssfang — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # Release History
 
 ## 5.1.0-beta.4 (Unreleased)
+### New features
+- Added property `Length` to `CategorizedEntity`, `SentenceSentiment`, `LinkedEntityMatch`, `AspectSentiment`, `OpinionSentiment`, and `PiiEntity`.
```
Comment (mssfang) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:5]: Make sure the healthcare entity has the "length" property as well.

**Pull request merged**